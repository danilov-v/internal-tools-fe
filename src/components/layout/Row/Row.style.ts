import styled from 'styled-components';

type ContainerProps = {
  justify:
    | 'center'
    | 'flex-end'
    | 'flex-start'
    | 'space-around'
    | 'space-between';
  mb: number;
  ml: number;
  mr: number;
  mt: number;
  pb: number;
  pl: number;
  pr: number;
  pt: number;
};

export const Container = styled.div<ContainerProps>`
  display: flex;
  justify-content: ${({ justify }) => justify};
  margin-bottom: ${({ mb }) => mb}px;
  margin-left: ${({ ml }) => ml}px;
  margin-right: ${({ mr }) => mr}px;
  margin-top: ${({ mt }) => mt}px;
  padding-bottom: ${({ pb }) => pb}px;
  padding-left: ${({ pl }) => pl}px;
  padding-right: ${({ pr }) => pr}px;
  padding-top: ${({ pt }) => pt}px;
`;
