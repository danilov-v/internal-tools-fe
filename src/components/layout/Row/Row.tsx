import React from 'react';

import * as S from './Row.style';

const Row: React.FC<RowProps> = ({
  justify = 'flex-start',
  children = '',
  className = '',
  mb = 8,
  ml = 0,
  mr = 0,
  mt = 8,
  pb = 0,
  pl = 0,
  pr = 0,
  pt = 0,
}) => (
  <S.Container
    className={className}
    justify={justify}
    mb={mb}
    ml={ml}
    mr={mr}
    mt={mt}
    pb={pb}
    pl={pl}
    pr={pr}
    pt={pt}
  >
    {children}
  </S.Container>
);

type RowProps = {
  className?: string;
  justify?:
    | 'center'
    | 'flex-end'
    | 'flex-start'
    | 'space-around'
    | 'space-between';
  mb?: number;
  ml?: number;
  mr?: number;
  mt?: number;
  pb?: number;
  pl?: number;
  pr?: number;
  pt?: number;
};

export { Row };
