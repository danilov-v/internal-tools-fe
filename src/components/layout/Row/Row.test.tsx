import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import 'jest-styled-components';

import { Row } from './Row';

const getComponent = (props = {}): RenderResult => {
  const parsedProps = {
    children: '',
    ...props,
  };

  return render(<Row {...parsedProps} />);
};

describe('Row component', () => {
  it('matches snapshot with only required props', () => {
    expect(getComponent().baseElement).toMatchSnapshot();
  });

  it('matches snapshot with additional props', () => {
    const component = getComponent({
      justify: 'center',
      children: 'children',
      mb: 100,
      pt: 20,
    });

    expect(component.baseElement).toMatchSnapshot();
  });
});
