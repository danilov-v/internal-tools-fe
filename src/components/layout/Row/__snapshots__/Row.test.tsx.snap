// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component matches snapshot with additional props 1`] = `
<body>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 100px;
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 8px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 20px;
}

<div>
    <div
      class="c0"
    >
      children
    </div>
  </div>
</body>
`;

exports[`Row component matches snapshot with only required props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 8px;
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 8px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
}

<body>
  <div>
    <div
      class="c0"
    />
  </div>
</body>
`;
